### PR TITLE
Fix: Execution of untrusted code when cloning Git submodule (CVE-2024-32002) (#6352)

### LINKED ISSUE #6352: Execution of untrusted code when cloning Git submodule (CVE-2024-32002)
https://nvd.nist.gov/vuln/detail/CVE-2024-32002

From the inspector findings sheet:

https://docs.google.com/spreadsheets/d/1RWF7g5wRKWPGovLw4jpJGX_XMi8aWLXLOvvE5rxqgH8/edit#gid=1268375573

image

This repository uses [easyrsa](https://github.com/OpenVPN/easy-rsa) as a submodule. If an exploit was injected there, we would potentially execute malicious code during checkout on GitLab and locally.

Images currently affected are

- docker.io/gitlab/gitlab-runner/ubuntu-v16.11.1-linux-amd64
- docker.io/gitlab/gitlab-runner/ubuntu-v16.11.1-linux-arm64
- docker.io/library/python/3.11.9-bullseye-linux-amd64
- docker.io/library/python/3.11.9-bullseye-linux-arm64
- docker.io/ucscgi/azul-pycharm/2024.1.2-24-linux-amd64
- docker.io/ucscgi/azul-pycharm/2024.1.2-24-linux-arm64

The `gitlab-runner` images and `python` images are used pervasively, locally and on GitLab.

We don't use the `azul-pycharm` image in conjunction with cloning a repository or checking out submodules so the severity there is lower. The fix there could be to simply remove any `git` binaries from the image.

### PR DESCRIPTION
**Pull request by dsotirho-ucsc**
Connected issues: #6352

### Author

- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] On ZenHub, PR is connected to all issues it (partially) resolves
- [x] PR description links to connected issues
- [x] PR title matches<sup>1</sup> that of a connected issue <sub>or comment in PR explains why they're different</sub>
- [x] PR title references all connected issues
- [x] For each connected issue, there is at least one commit whose title references that issue

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title

### Author (partiality)

- [x] Added `p` tag to titles of partial commits
- [x] This PR is labeled `partial` <sub>or completely resolves all connected issues</sub>
- [x] This PR partially resolves each of the connected issues <sub>or does not have the `partial` label</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain <sub>or is not chained to another PR</sub>
- [x] The blocking PR is labeled `base` <sub>or this PR is not chained to another PR</sub>
- [x] This PR is labeled `chained` <sub>or is not chained to another PR</sub>

### Author (reindex, API changes)

- [x] Added `r` tag to commit title <sub>or the changes introduced by this PR will not require reindexing of any deployment</sub>
- [x] This PR is labeled `reindex:dev` <sub>or the changes introduced by it will not require reindexing of `dev`</sub>
- [x] This PR is labeled `reindex:anvildev` <sub>or the changes introduced by it will not require reindexing of `anvildev`</sub>
- [x] This PR is labeled `reindex:anvilprod` <sub>or the changes introduced by it will not require reindexing of `anvilprod`</sub>
- [x] This PR is labeled `reindex:prod` <sub>or the changes introduced by it will not require reindexing of `prod`</sub>
- [x] This PR is labeled `reindex:partial` and its description documents the specific reindexing procedure for `dev`, `anvildev`, `anvilprod` and `prod` <sub>or requires a full reindex or carries none of the labels `reindex:dev`, `reindex:anvildev`, `reindex:anvilprod` and `reindex:prod`</sub>
- [x] This PR and its connected issues are labeled `API` <sub>or this PR does not modify a REST API</sub>
- [x] Added `a` (`A`) tag to commit title for backwards (in)compatible changes <sub>or this PR does not modify a REST API</sub>
- [x] Updated REST API version number in `app.py` <sub>or this PR does not modify a REST API</sub>

### Author (upgrading deployments)

- [x] Ran `make image_manifests.json` and committed the resulting changes <sub>or this PR does not modify `azul_docker_images`, or any other variables referenced in the definition of that variable</sub>
- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading deployments</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading deployments</sub>
- [x] This PR is labeled `upgrade` <sub>or does not require upgrading deployments</sub>
- [x] This PR is labeled `deploy:shared` <sub>or does not modify `image_manifests.json`, and does not require deploying the `shared` component for any other reason</sub>
- [x] This PR is labeled `deploy:gitlab` <sub>or does not require deploying the `gitlab` component</sub>
- [x] This PR is labeled `deploy:runner` <sub>or does not require deploying the `runner` image</sub>

### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any connected issues <sub>or the none of the stable branches (`anvilprod` and `prod`) have temporary hotfixes for any of the issues connected to this PR</sub>

### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not modify `requirements*.txt`, `common.mk`, `Makefile` and `Dockerfile`</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not modify `requirements*.txt`</sub>
- [x] This PR is labeled `reqs` <sub>or does not modify `requirements*.txt`</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not modify functionality that could affect the IT outcome</sub>

### Peer reviewer (after approval)

- [x] PR is not a draft
- [x] Ticket is in *Review requested* column
- [x] PR is awaiting requested review from system administrator
- [x] PR is assigned to only the system administrator

### System administrator (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issues as `demo` or `no demo`
- [x] Commented on connected issues about demo expectations <sub>or all connected issues are labeled `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] A comment to this PR details the completed security design review
- [x] PR title is appropriate as title of merge commit
- [x] `N reviews` label is accurate
- [x] Moved ticket to *Approved* column
- [x] PR is assigned to only the operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex:…` labels and `r` commit title tag
- [x] Checked that demo expectations are clear <sub>or all connected issues are labeled `no demo`</sub>
- [x] Squashed PR branch and rebased onto `develop`
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub
- [x] Ran `_select dev.shared && CI_COMMIT_REF_NAME=develop make -C terraform/shared apply_keep_unused` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select dev.gitlab && CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply` <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Ran `_select anvildev.shared && CI_COMMIT_REF_NAME=develop make -C terraform/shared apply_keep_unused` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select anvildev.gitlab && CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply` <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Checked the items in the next section <sub>or this PR is labeled `deploy:gitlab`</sub>
- [x] PR is assigned to only the system administrator <sub>or this PR is not labeled `deploy:gitlab`</sub>

### System administrator

- [x] Background migrations for `dev.gitlab` are complete <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Background migrations for `anvildev.gitlab` are complete <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] PR is assigned to only the operator

### Operator (before pushing merge the commit)

- [x] Ran `_select dev.gitlab && make -C terraform/gitlab/runner` <sub>or this PR is not labeled `deploy:runner`</sub>
- [x] Ran `_select anvildev.gitlab && make -C terraform/gitlab/runner` <sub>or this PR is not labeled `deploy:runner`</sub>
- [x] Added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `dev`</sub>
- [x] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvildev`</sub>
- [x] Started reindex in `sandbox` <sub>or this PR is not labeled `reindex:dev`</sub>
- [x] Started reindex in `anvilbox` <sub>or this PR is not labeled `reindex:anvildev`</sub>
- [x] Checked for failures in `sandbox` <sub>or this PR is not labeled `reindex:dev`</sub>
- [x] Checked for failures in `anvilbox` <sub>or this PR is not labeled `reindex:anvildev`</sub>
- [x] The title of the merge commit starts with the title of this PR
- [x] Added PR # reference to merge commit title
- [x] Collected commit title tags in merge commit title <sub>but only included `p` if the PR is also labeled `partial`</sub>
- [x] Moved connected issues to *Merged lower* column in ZenHub
- [x] Pushed merge commit to GitHub

### Operator (chain shortening)

- [x] Changed the target branch of the blocked PR to `develop` <sub>or this PR is not labeled `base`</sub>
- [x] Removed the `chained` label from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [x] Removed the blocking relationship from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [x] Removed the `base` label from this PR <sub>or this PR is not labeled `base`</sub>

### Operator (after pushing the merge commit)

- [x] Pushed merge commit to GitLab `dev`
- [x] Pushed merge commit to GitLab `anvildev`
- [x] Build passes on GitLab `dev`
- [x] Reviewed build logs for anomalies on GitLab `dev`
- [x] Build passes on GitLab `anvildev`
- [x] Reviewed build logs for anomalies on GitLab `anvildev`
- [x] Ran `_select dev.shared && make -C terraform/shared apply` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select anvildev.shared && make -C terraform/shared apply` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Deleted PR branch from GitHub
- [x] Deleted PR branch from GitLab `dev`
- [x] Deleted PR branch from GitLab `anvildev`

### Operator (reindex)

- [x] Deindexed all unreferenced catalogs in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Deindexed all unreferenced catalogs in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Deindexed specific sources in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Deindexed specific sources in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Indexed specific sources in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Indexed specific sources in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Started reindex in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Started reindex in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Checked for, triaged and possibly requeued messages in both fail queues in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Checked for, triaged and possibly requeued messages in both fail queues in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Emptied fail queues in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Emptied fail queues in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>

### Operator

- [x] Propagated the `deploy:shared`, `deploy:gitlab`, `deploy:runner`, `API`, `reindex:partial`, `reindex:anvilprod` and `reindex:prod` labels to the next promotion PRs <sub>or this PR carries none of these labels</sub>
- [x] Propagated any specific instructions related to the `deploy:shared`, `deploy:gitlab`, `deploy:runner`, `API`, `reindex:partial`, `reindex:anvilprod` and `reindex:prod` labels, from the description of this PR to that of the next promotion PRs <sub>or this PR carries none of these labels</sub>
- [x] PR is assigned to no one

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Comment by dsotirho-ucsc** [comment]
Using `-slim-bullseye` instead of `-slim` (bookworm) due to Debian bookworm incompatibility with FIPS mode:
- `fatal error in libgcrypt` during build with `-slim`
  - https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/96127
- Existing ticket `debian bookworm apt-get update fails with fatal error in libgcrypt`
  - https://www.github.com/debuerreotype/docker-debian-artifacts/issues/202
- Further discussion on the FIPS issue with Debian bookworm
  - https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1014517

PR (with `-slim-bullseye`) was successfully tested on GitLab `anvildev`
- Attempt 1: Failed due to missing requirements update commit https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/pipelines/18295
- Attempt 2: Success https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/pipelines/18296

**Comment by dsotirho-ucsc** [comment]
[6405_IT_2024-07-25.txt](https://github.com/user-attachments/files/16381246/6405_IT_2024-07-25.txt)

**Comment by dsotirho-ucsc** [comment]
[6405_IT_2024-07-30.txt](https://github.com/user-attachments/files/16434738/6405_IT_2024-07-30.txt)

**Review by achave11-ucsc — DISMISSED** [review]
Approved ✅

File: Dockerfile [hunk 1]
```diff
@@ -12,6 +12,10 @@ SHELL ["/bin/bash", "-c"]
 ARG azul_image_version=1
 RUN apt-get update && apt-get upgrade -y
 
+RUN apt-get -y install build-essential
+RUN apt-get -y install curl
+RUN apt-get -y install unzip
```
Comment (hannes-ucsc) [inline Dockerfile]: All four commands should be done in one RUN. I can explain in PL if you want.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Comment by dsotirho-ucsc** [comment]
[6405_IT_2024-08-01.txt](https://github.com/user-attachments/files/16463749/6405_IT_2024-08-01.txt)

Requirements update log to show success of step [2/12]:
```
$ make requirements_update
git restore requirements.trans.txt requirements.dev.trans.txt
perl -i -p -e 's/^(?!#)/#/' requirements.trans.txt requirements.dev.trans.txt
DOCKER_DEFAULT_PLATFORM=linux/amd64 /Library/Developer/CommandLineTools/usr/bin/make docker_deps docker_dev_deps
docker build --build-arg azul_docker_registry= --build-arg azul_python_image=docker.io/library/python:3.11.9-slim-bullseye --build-arg azul_docker_version=27.0.3 --build-arg azul_terraform_version=1.9.2 --build-arg PIP_DISABLE_PIP_VERSION_CHECK=1 --build-arg make_target=requirements_runtime_deps --tag docker.gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/deps:latest .
[+] Building 103.7s (18/18) FINISHED                                                                                                                  docker:desktop-linux
 => [internal] load build definition from Dockerfile                                                                                                                  0.0s
 => => transferring dockerfile: 2.85kB                                                                                                                                0.0s
 => [internal] load metadata for docker.io/library/python:3.11.9-slim-bullseye                                                                                        1.2s
 => [auth] library/python:pull token for registry-1.docker.io                                                                                                         0.0s
 => [internal] load .dockerignore                                                                                                                                     0.0s
 => => transferring context: 173B                                                                                                                                     0.0s
 => [ 1/12] FROM docker.io/library/python:3.11.9-slim-bullseye@sha256:c54ab9db1eaf3af9f171c2add48b1e7d409738f3345eed9207d303c7c2e4b7b5                                6.2s
 => => resolve docker.io/library/python:3.11.9-slim-bullseye@sha256:c54ab9db1eaf3af9f171c2add48b1e7d409738f3345eed9207d303c7c2e4b7b5                                  0.0s
 => => sha256:d30714b439e5fdb8ca413012963bc1cdc8e257a2676f1aa7a6913e4785c78f9a 6.89kB / 6.89kB                                                                        0.0s
 => => sha256:5de87e84afeec60e41fb003112c283b04a73e50c8d579c88bd21d668fd688485 31.43MB / 31.43MB                                                                      5.0s
 => => sha256:6bc82b1e04f1ddd39e4b5c16fbbba3ff67191879e522a7c24471359c276f05de 1.08MB / 1.08MB                                                                        0.7s
 => => sha256:8eabeab7b7afff214128063e71100864e670caf96dfe7f7dae20a259ba4a401f 12.02MB / 12.02MB                                                                      3.6s
 => => sha256:c54ab9db1eaf3af9f171c2add48b1e7d409738f3345eed9207d303c7c2e4b7b5 9.12kB / 9.12kB                                                                        0.0s
 => => sha256:f0bbbbd407325d3a5b5b6ea37cb336f6e9770a2377e9a9b825fcebbc532745cc 1.94kB / 1.94kB                                                                        0.0s
 => => sha256:0133c99fe7ec69da54df8804d15da4c90d711bbd97bde69b28f7979d9ff32b86 232B / 232B                                                                            0.8s
 => => sha256:35191651da7f76d3580e6bd9c8ae4317d1b6830aca72a29c08387085796b8a02 3.21MB / 3.21MB                                                                        1.6s
 => => extracting sha256:5de87e84afeec60e41fb003112c283b04a73e50c8d579c88bd21d668fd688485                                                                             0.8s
 => => extracting sha256:6bc82b1e04f1ddd39e4b5c16fbbba3ff67191879e522a7c24471359c276f05de                                                                             0.0s
 => => extracting sha256:8eabeab7b7afff214128063e71100864e670caf96dfe7f7dae20a259ba4a401f                                                                             0.2s
 => => extracting sha256:0133c99fe7ec69da54df8804d15da4c90d711bbd97bde69b28f7979d9ff32b86                                                                             0.0s
 => => extracting sha256:35191651da7f76d3580e6bd9c8ae4317d1b6830aca72a29c08387085796b8a02                                                                             0.1s
 => [internal] load build context                                                                                                                                     0.6s
 => => transferring context: 3.05kB                                                                                                                                   0.6s
 => [ 2/12] RUN apt-get update     && apt-get upgrade -y     && apt-get -y install build-essential curl unzip                                                        32.7s
 => [ 3/12] RUN curl -o /usr/bin/docker-credential-ecr-login     https://amazon-ecr-credential-helper-releases.s3.us-east-2.amazonaws.com/0.7.0/linux-amd64/docker-c  1.4s
 => [ 4/12] RUN mkdir -p ${HOME}/.docker     && printf '{"credHelpers": {"%s": "ecr-login"}}\n' ""     > "${HOME}/.docker/config.json"                                0.1s
 => [ 5/12] RUN mkdir terraform     && (set -o pipefail         && cd terraform         && curl -s -o terraform.zip            https://releases.hashicorp.com/terraf  3.5s
 => [ 6/12] RUN install -m 0755 -d /etc/apt/keyrings                                                                                                                  0.1s
 => [ 7/12] COPY --chmod=0644 bin/keys/docker-apt-keyring.pgp /etc/apt/keyrings/docker.gpg                                                                            0.0s
 => [ 8/12] RUN set -o pipefail     && (       echo "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/l  41.0s
 => [ 9/12] RUN mkdir /build                                                                                                                                          0.1s
 => [10/12] WORKDIR /build                                                                                                                                            0.0s
 => [11/12] COPY requirements*.txt common.mk Makefile ./                                                                                                              0.0s
 => [12/12] RUN make virtualenv     && source .venv/bin/activate     && make requirements_runtime_deps     && rm requirements*.txt common.mk Makefile                16.3s
 => exporting to image                                                                                                                                                0.8s
 => => exporting layers                                                                                                                                               0.8s
 => => writing image sha256:c78b6abc8ad9f599efd4ff013e233e616aabb6d9c32ae659b10a7e608d22c1a9                                                                          0.0s
 => => naming to docker.gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/deps:latest                                                                                       0.0s

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/v6xfec11madsml7m6kszel2x7
```

**Comment by dsotirho-ucsc** [comment]
[6405_IT_2024-08-05.txt](https://github.com/user-attachments/files/16501998/6405_IT_2024-08-05.txt)

**Comment by hannes-ucsc** [comment]
Both of your PRs referring to this issue are partial. That would imply that the issue is not completely resolved, even after both of the PR are merged. I don't remember asking for partial PRs. LMK if I did. One of the PRs would have to be partial, the other one not. That then introduces an ordering constraint in which these PRs would have to be merged, partial PR first, then the other. That again seems convoluted. Please combine both PRs.

**Comment by dsotirho-ucsc** [comment]
[6405_IT_2024-08-13.txt](https://github.com/user-attachments/files/16603833/6405_IT_2024-08-13.txt)

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Conflicts, actually.

**Comment by dsotirho-ucsc** [comment]
The update of the PyCharm image originally part of this PR has already been merged into `develop` by PR #6490

**Comment by dsotirho-ucsc** [comment]
[6405_IT_2024-08-15.txt](https://github.com/user-attachments/files/16627896/6405_IT_2024-08-15.txt)

**Review by hannes-ucsc — DISMISSED** [review]

**Comment by hannes-ucsc** [comment]
### Security design review

- [x] Security design review completed; this PR does **not** …
  - [x] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with Terra 
  - [x] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [x] … affect the permissions of external users like access to
    - TDR snapshots
  - [x] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [x] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [x] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - ~Docker images~ https://github.com/DataBiosphere/azul/pull/6405#issuecomment-2291905352
    - Terraform providers
  - [x] … add an interface that exposes sensitive or confidential data at the security boundary
  - [x] … affect the encryption of data at rest
  - [x] … require persistence of sensitive or confidential data that might require encryption at rest
  - [x] … require unencrypted transmission of data within the security boundary
  - [x] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [x] Documentation on any unchecked boxes is provided in comments below

**Comment by hannes-ucsc** [comment]
Security design review: This change removes unused dependencies in order to eliminate vulnerabilities found in them.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**